src/index.js

```diff
@@ -3,6 +3,10 @@ import {newInstance} from './wasm';
 export default class SHA256 {
   constructor() {
     this.ctx = newInstance();
+    this.wasmInputValue = this.ctx.input.value;
+    this.wasmOutputValue = this.ctx.output.value;
+    this.inputArray = new Uint8Array(this.ctx.memory.buffer, this.wasmInputValue, this.ctx.INPUT_LENGTH);
```

Comment:
since you are already at it, you could also potentially pass this optimization benefit to the update function as well and save the new call there as well :smiley: